openslides_backend/action/actions/motion_comment/create_delete_update.py

```diff
@@ -1,12 +1,76 @@
+from typing import Any, Dict
+
 from ....models.models import MotionComment
+from ....permissions.permissions import OrganisationManagementLevel, Permissions
+from ....services.datastore.commands import GetManyRequest
+from ....shared.exceptions import PermissionDenied
+from ....shared.patterns import Collection, FullQualifiedId
+from ...action import Action
 from ...action_set import ActionSet
+from ...generics.delete import DeleteAction
+from ...generics.update import UpdateAction
 from ...mixins.create_action_with_inferred_meeting import (
-    get_create_action_with_inferred_meeting,
+    CreateActionWithInferredMeeting,
 )
 from ...util.default_schema import DefaultSchema
 from ...util.register import register_action_set
 
 
+class PermissionMixin(Action):
+    def check_permissions(self, instance: Dict[str, Any]) -> None:
+        super().check_permissions(instance)
+
+        # check for superuser
+        if self.user_id > 0:
+            user = self.datastore.get(
+                FullQualifiedId(Collection("user"), self.user_id),
+                [
+                    "organisation_management_level",
+                ],
+            )
+            if (
+                user.get("organisation_management_level")
+                == OrganisationManagementLevel.SUPERADMIN
+            ):
+                return
+
+        # get section_id, create vs delete/update case.
+        if "section_id" in instance:
+            section_id = instance["section_id"]
+        else:
+            comment = self.datastore.get(
+                FullQualifiedId(self.model.collection, instance["id"]), ["section_id"]
+            )
+            section_id = comment["section_id"]
+        section = self.datastore.get(
+            FullQualifiedId(Collection("motion_comment_section"), section_id),
+            ["write_group_ids"],
+        )
+        gmr = GetManyRequest(
+            Collection("group"), section["write_group_ids"], ["user_ids"]
+        )
```

Comment:
It's better to never compare user ids explicitely, since these can potentially be a huge number. So I would prefer to fetch the `group_ids` from the request user and check if the intersection of these with the `write_group_ids` is empty or not.

Reply:
Okay, done.